feat: Category screens

Lantum-Brendan opened the pull request:
Implement category screens

nfebe commented on components/categories/CategoryList.vue:
This should be generic list component, that can be used to list other things. Otherwise do you want to create a list type for every model?

nfebe commented on components/categories/CategoryIconPicker.vue:
The icon picker is not just used for categories, so it should be a generic component `IconPicker`,  also there should be a better way of loading the lucid-icons, instead of importing them like that?

nfebe commented on components/categories/CategoryEmptyState.vue:
`EmptyState` is a generic component and would be used in other places so remove it from here and simply use it as `EmptyState` instead of `CategoryEmptyState`

nfebe reviewed (changes requested):
See inline comments.

Lantum-Brendan commented:
##Changes made: 

1. Changed components/categories to components/contentSharedPages
2. Renamed the files and removed most of the hardcoded data
3. Did not use TTopCard: instead maintained categoryTopCard now ContentTopCard: reasoning, the content of TTopCard is hard coded. contentTopCard takes the pagename dynamically directly from the page.(if permitted i should refactor the TTopCard like the contentTopcard so it is reusable across all pages and dynamically rendering the page name with appropriate buttons)
4. CategoryIconPicker and CategoryEmptyState changed to generic. --> now iconpicker imports all icons and not each icon imported as before.
5. Removed the unused phonenumber validation logic  and imports from Register.vue page

- Tested the re-usability of the new changes by adding the Groups page.  
- can modify the form to add more fields and optionally display fields based on the page we are on. Reuse across the wallets/parties pages made possible

nfebe commented:
> Changed components/categories to components/contentSharedPages

Not needed, the idea behind components is to self-contain things but also to share them, once things are already in components a name like `contentSharedPages` as a sub folder does not make any sense. `components/EmptState` `components/ContentTable`, `component/ContentList` works just fine.

> Did not use TTopCard: instead maintained categoryTopCard now ContentTopCard: reasoning, the content of TTopCard is hard coded. contentTopCard takes the pagename dynamically directly from the page.(if permitted i should refactor the TTopCard like the contentTopcard so it is reusable across all pages and dynamically rendering the page name with appropriate buttons)

That is an opportunity the make `TTopCard` reusable, you can duplicate code with the excuse that some component has hard-coded the data, there is not API integration yet so that is expected.

>  Removed the unused phonenumber validation logic and imports from Register.vue page

Your PR has nothing to do with registration, if you touched any file there checkout the file as that would be an error much like that one that led you to having crazing merge conflicts I had to fix.

nfebe commented on components/contentSharedPages/ContentForm.vue:
```diff
@@ -0,0 +1,202 @@
+<template>
+  <form class="entity-form" @submit.prevent="handleSubmit">
```
I don't think a form is a good example of a re-usable component. This is overkill. Mantain `categories/AddCategoryForm` the base elements (like icon pickers) can be reusable not the form themselves.

Lantum-Brendan replied:
Changes made

1. Removed categoriesTopCard as requested and made the TTopCard reusable
2. Separated forms; no longer one form being reused
3. Moved shared files from categories/ to components/

nfebe commented on components/contentSharedPages/ContentTopCard.vue:
```diff
@@ -0,0 +1,106 @@
+<template>
```
NO! Use `TTopCard` don't reinvent the wheel

kofimokome commented:
I think we should start thinking of how to make the app support multiple languages. It must not be done in this PR.  It can be moved to a separate issue.

kofimokome commented on components/categories/CategoryForm.vue:
```diff
@@ -0,0 +1,70 @@
+
+function handleSubmit() {
+  iconError.value = !form.value.icon;
+  if (!form.value.name || iconError.value) return;
```
Ensure `form.value.name.trim()` is not an empty string or has a length greater than 0 before saving the category.

Filling a string like "        " bypasses your checks
